Add e2e tests for the host details view

rtorrero opened the pull request:
This is another PR that introduces end to end tests for the host details view. I've written the Gherkin from the test scenarios described by @abravosuse 

I followed the document strictly where only a few aspects of the view are checked, such as the  SAP System SID or cluster name, but I'd like see what you think and if we should check everything or just a few things here.

Also notice that in the `Background` section on the `host_details.feature` I describe the whole 27-hosts deployment as I'm reusing it from the hosts overview tests. I'm unsure if we should only mention in the background the aspects of the fixtures that we use and how it would look.

abravosuse commented:
@rtorrero awesome! How about we stay together tomorrow after the daily standup and you show me exactly what the test covers? I've just noticed that the links to Azure components (vm and resource group) are no longer enabled. Therefore yes, the test would simply consist on checking the SAP system link and the cluster link.

nelsonkopliku commented on test/e2e/cypress/fixtures/host-details/host_details.feature:
```diff
@@ -0,0 +1,48 @@
+
+Background:
+    Given an healthy SAP deployment of 27 hosts having following agent ids and hostnames
+    # a09d9cf3-46c1-505c-8fb8-4b0a71a9114e => vmdrbdprd01
```
Let's consider if we need this list or the simple detail of that one host we're testing against :wink:

nelsonkopliku commented on test/e2e/cypress/integration/host_details.js:
```diff
@@ -0,0 +1,48 @@
+
+    describe('Detailed view for a specific host should be available', () => {
+        it('should show the host I clicked on in the overview', () => {
+            cy.get('.border-top > :nth-child(1) > .col-sm-12 > .row > :nth-child(1) > .text-muted').should('contain', 'vmhdbprd01')
```
consider whether `vmhdbprd01`, `1154f7678ac587e5f0f242830a5201f1` or other hardcoded values here might be part of the imported `selected_host`

nelsonkopliku reviewed (commented):
Great Job so far @rtorrero! Really happy to see how well you onboarded on this! :rocket: 
Just a couple of comments that I hope might be useful.

rtorrero commented on test/e2e/cypress/fixtures/host-details/selected_host.js:
```diff
@@ -0,0 +1,13 @@
+    resourceGroup: 'resourceGroupName',
+}
+
+export const agentId = () => selectedHost.agentId;
```
Yep, I was initially following the style from your test, but this quickly turned into a mess as I added more stuff on the selected_host.js file. Fixed!

rtorrero commented:
I've added some additional checks requested by @abravosuse and adjusted the code a bit to follow @nelsonkopliku suggestions.

Should be good now but let me know if I missed anything

nelsonkopliku commented on test/e2e/cypress/fixtures/host-details/host_details.feature:
```diff
@@ -0,0 +1,27 @@
+        Given I am in the host details view (/hosts/9cd46919-5f19-59aa-993e-cf3736c71053)
+        Then the displayed details should include a link to a SAP system details view with SID equal to HDP
+        And a link to a cluster details view with name `hana_cluster` should be under the cluster label
+        And one entry with ID 14f65ee5b3a07a320272b47c7c5d4be6 should be present in the SAP instances list
```
BTW, where does this `14f65ee5b3a07a320272b47c7c5d4be6` come from, I don't see it used anywhere else :thinking:

arbulu89 commented on test/e2e/cypress/integration/host_details.js:
```diff
@@ -0,0 +1,67 @@
+        selectedHost.sapInstanceId
+      );
+    });
+    it(`should show a link to the cluster details view for ${selectedHost.clusterName}`, () => {
```
This test looks like belongs to other `describe`, something about clusters

nelsonkopliku commented on test/e2e/cypress/integration/host_details.js:
```diff
@@ -0,0 +1,67 @@
+    cy.resetDatabase();
+    cy.loadScenario("healthy-27-node-SAP-cluster");
+
+    const agents = new Array();
```
It could be simplified to `const agents = [];`

nelsonkopliku commented on test/e2e/cypress/integration/host_details.js:
```diff
@@ -0,0 +1,67 @@
+
+    const agents = new Array();
+    agents.push(selectedHost.agentId);
+    cy.task("startAgentHeartbeat", agents);
```
Or simply `cy.task("startAgentHeartbeat", [selectedHost.agentId]);`

arbulu89 reviewed (commented):
Hey @rtorrero !

Maybe they are planned for future PRs but, I would try to include the:
- SUSE subscriptions table test
- Trento agent status test
- The `Hosts > vmhdbprd01` text test (just after the title)
- Include some `Not found` test, but I guess this PR only includes happy path tests

If they are not coming in this PR, everything else looks pretty good!

nelsonkopliku commented on test/e2e/cypress/integration/host_details.js:
```diff
@@ -0,0 +1,67 @@
+  describe("Detailed view for a specific host should be available", () => {
+    it("should show the host I clicked on in the overview", () => {
+      cy.get(
+        ".border-top > :nth-child(1) > .col-sm-12 > .row > :nth-child(1) > .text-muted"
```
I'd consider adding some more specific css class in order to simplify selector. Something like a container `tn-host-detail-container` and then the name in a child element like `tn-hostname`

That would simplify selecting `.tn-host-detail-container > .tn-hostname`

What do you think?

nelsonkopliku reviewed (approved):
LGTM! Just a couple of minor comments.

dottorblaster reviewed (approved):
Addressing comments by @arbulu89 and @nelsonkopliku should be more than enough. Brilliant! LGTM

rtorrero commented on test/e2e/cypress/fixtures/host-details/host_details.feature:
```diff
@@ -0,0 +1,27 @@
+        # resourceGroup: 'resourceGroupName',
+        And a Trento installation on this host with the following agent ID
+    # agentId: '9cd46919-5f19-59aa-993e-cf3736c71053'
+
```
True, I added it in the host_details but forgot to mention it in the feature. Nice catch :-)

rtorrero commented on test/e2e/cypress/integration/host_details.js:
```diff
@@ -0,0 +1,67 @@
+    agents.push(selectedHost.agentId);
+    cy.task("startAgentHeartbeat", agents);
+
+    cy.visit("/");
```
I've improved it a bit, sorry if it didn't make much sense, my JS is a bit rusty!

rtorrero commented:
Everything has been addressed with the exception that as you suggested, we are going for happy path only in this tests. We don't want to get too deep in testing all elements of the view as we expect that lots of these things are going to change soon; at least that's what we agreed with @abravosuse

nelsonkopliku reviewed (approved):
Great job @rtorrero! :rocket: 

There's still some selectors that might be improved application wide, which would require refactoring different views and related tests, but we can address those later by iterating.

For me it is a green light.
